checker/src/main/java/org/checkerframework/checker/resourceleak/MustCallInference.java

```diff
@@ -122,12 +142,19 @@ public class MustCallInference {
   /** The element for the current method. */
   private final ExecutableElement methodElt;
 
-  /** The ClassTree referring to the enclosing class of the current method. */
+  /** The tree for the enclosing class of the current method. */
   private final ClassTree classTree;
 
-  /** The element referring to the enclosing class of the current method. */
+  /** The element for the enclosing class of the current method. */
   private final @Nullable TypeElement classElt;
 
+  /**
+   * This map is used to track must-alias relationships between return nodes and method parameters.
+   * The keys are the obligation of return nodes, and the values are the index of current method
+   * formal parameter (1-based) that is aliased with the return node.
+   */
+  private final Map<Obligation, Integer> returnNodeToParameter = new HashMap<>();
```

Comment:
Should this be named "returnObligationToParameter"?

More generally, please discuss the choice to use obligations rather than nodes as the keys.  The documentation says that the map is from return nodes to method parameters, but the implementation is different.  Why are obligations needed?  Is it because they indicate resource aliasing relationships?  Or is it for some other reason?

Reply:
They are needed to check aliasing relationships of return nodes with parameters.